airflow/models/dag.py

```diff
@@ -2028,7 +2027,7 @@ class DagModel(Base):
     fileloc = Column(String(2000))
     # The base directory used by Dag Processor that parsed this dag.
     processor_subdir = Column(String(2000), nullable=True)
-    bundle_id = Column(UUIDType(binary=False), ForeignKey("dag_bundle.id"), nullable=True)
+    bundle_name = Column(StringID(), ForeignKey("dag_bundle.name"), nullable=True)
```

Comment:
I think to preserve history, we should use an association table so that when a dag object is assigned a new bundle object, the history will be preserved. Example: If a dag 'A' is in dag-bundle 'DA', and 'DA' is no longer configured or the name was changed, which triggers a new dag-bundle object, say 'DB', which now has dag 'A' in it. The DAG bundle_name will update to the new dagbundle object 'DB', causing us to lose the previous bundle name. With an association table, we can have an is_active in the table that tells whether the bundle has been removed. However, there will be more complex queries. 

Another thing I thought of is using a history table like in TIH, but DAG changes more often.

Reply:
The association table would have to also have a 'time' component - just knowing it was, at one point, associated with another bundle isn't that interesting. I kinda view the bundle similarly to the "fileloc" - the DAG itself is the primary thing, the rest is just supplemental. The `enabled` column on bundles today is enough to avoid cascading deletes.

I do think having complete history of the `dag` table could be useful, but I don't want to take that on with these changes.